source/controlTypes/role.py

```diff
@@ -177,6 +177,9 @@ def _displayStringLabels(self):
 	MARKED_CONTENT = 153
 	BUSY_INDICATOR = 154  # Used for progress bars with indeterminate state
 	# To maintain backwards compatibility, above Roles must maintain their values.
+	COMMENT = auto()

```

Comment:
What exact backwards compatibility guarantees we offer add-on developers with regard to numeric values of roles? If we guarantee only that all roles prior to 2022.1 maintain their integer values then using `auto` here is fine. On the other hand since roles are not numbered sequentially I can imagine a case in which the newly added roles are assigned a value which is currently not taken but is in the middle of other roles, and some add-on relies on the missing integers for example by checking roles with less than or greater than operator.

Reply:
>What exact backwards compatibility guarantees we offer add-on developers with regard to numeric values of roles?

Given the changes that are already in 2022.1, I don't think we can make any guarantees in regards to numeric values of roles. The changes in 2022.1 dropped support for numeric values of roles, but kept aliases for ease of use. 

>some add-on relies on the missing integers for example by checking roles with less than or greater than operator

This is unsupported. Add-on authors should not be relying on this to begin with. Roles could be created at any point which used "missing integers". Using "missing integers" would also cause conflict with other add-ons. Using Enums discourages these approaches by design. The NVDA API doesn't have "reserved space" for add-ons as it stands.

Reply:
@feerrenrut - thoughts?

Reply:
@lukaszgo1 - we are going to err on the side of caution here in trying to minimize introducing a breaking change. I've added some notes to clarify the position in bf4e4b2.

As per the notes and earlier comment, it is difficult to guarantee full support for treating `Role`s as integers.

Reply:
We've discussed this, @seanbudd will revert the use of `enum.auto` and add some comments to the Role enum to serve as future reference for the decisions we made around this.